src/chronify/time_series_mapper_representative.py

```diff
@@ -0,0 +1,140 @@
+            self._check_source_table_has_time_zone()
+        # TODO: add interval type handling (note annual has no interval type)
+
+        map_table_name = "map_table"
+        dfm = self._create_mapping(is_tz_naive)
+        if map_table_name in self._metadata.tables:
+            msg = (
+                f"table {map_table_name} already exists, delete it or use a different table name."
+            )
+            raise TableAlreadyExists(msg)
+        try:
+            with self._engine.connect() as conn:
+                write_database(dfm, conn, map_table_name, self._to_schema.time_config)
+                conn.commit()
+            self._metadata.reflect(self._engine, views=True)
+            self._apply_mapping(map_table_name)
+
+            mapped_table = Table(self._to_schema.name, self._metadata)
+            with self._engine.connect() as conn:
+                try:
+                    check_timestamps(conn, mapped_table, self._to_schema)
+                except Exception as e:
+                    print(e)
+                    print(f"check_timestamps failed, dropping mapped table {self._to_schema.name}")
+                    conn.execute(text(f"DROP TABLE {self._to_schema.name}"))
+        finally:
+            with self._engine.connect() as conn:
+                conn.execute(text(f"DROP TABLE {map_table_name}"))
+
+    def _create_mapping(self, is_tz_naive) -> pd.DataFrame:
+        """Create mapping dataframe"""
+        timestamp_generator = make_time_range_generator(self._to_schema.time_config)
+
+        to_time_col = self._to_schema.time_config.time_column
+        dft = pd.Series(timestamp_generator.list_timestamps()).rename(to_time_col).to_frame()
+        if is_tz_naive:
+            dfm = self._generator.create_tz_naive_mapping_dataframe(dft, to_time_col)
+        else:
+            with self._engine.connect() as conn:
+                table = Table(self._from_schema.name, self._metadata)
+                stmt = (
+                    select(table.c["time_zone"])
+                    .distinct()
+                    .where(table.c["time_zone"].is_not(None))
+                )
+                time_zones = read_database(stmt, conn, self._from_schema.time_config)[
+                    "time_zone"
+                ].to_list()
+            dfm = self._generator.create_tz_aware_mapping_dataframe(dft, to_time_col, time_zones)
+        return dfm
+
+    def _apply_mapping(self, map_table_name: str):
+        """Apply mapping to create result as a table according to_schema"""
+        left_table = Table(self._from_schema.name, self._metadata)
+        right_table = Table(map_table_name, self._metadata)
+        left_table_columns = [x.name for x in left_table.columns]
+        right_table_columns = [x.name for x in right_table.columns]
+
+        final_cols = set(self._to_schema.list_columns())
+        left_cols = set(left_table_columns).intersection(final_cols)
+        right_cols = final_cols - left_cols
```

Comment:
Is this guaranteed to be the same as `set(right_table_columns).intersection(final_cols)`?

Reply:
No, that's not what we want. We want `set(right_table_columns).intersection(final_cols - left_cols)`, but the intersection part is not needed because if any of the right_cols are not available in right_table_columns, table.c[col] in the next lines will yell.